### PR TITLE
Fix bundle

### LINKED ISSUE #5863: AI can't tell which doors are shocked without opening their menus
If an AI is subverted, shocks a lot of doors, and is then fixed, it's hard to then unelectrify all those doors. There's no exterior indicator to an AI if a door is electrified, so it has to click every door in an area it's electrified to see if it's electrified. This takes awhile, especially with lag making you wonder if the window has refreshed to reflect the new airlock you've clicked or not and having to stop and follow laws. It's easy to forget which areas you've shocked or miss a door, which leads to further harm or deaths, which leads to further accusations the AI is rogue. If the AI had a way to alt+click doors and know whether or not they've been electrified or unelectrified (such as a message in the chat box) it'd make the process much simpler.

### LINKED ISSUE #7389: Supermatter Shards are an administrative nightmare
They're far too easy to grief with with insufficient logging to help us catch the culprit. Mobs get deleted when they touch the shard so they lose the evidence that was stored there too. Most of this doesn't get reported to us until at least a few people are already dead to it.

My wish list for better logging tools:

- an announcement about who ordered it to the admins.
- When people are killed by it, admins should know 
  - who died
  - who touched the shard last and when
  - Who pushed the victim last
  - When they were pushed, if possible

### LINKED ISSUE #7839: More verbose Alert system
From this thread: https://tgstation13.org/phpBB/viewtopic.php?f=9&t=2800

Malkevin assembled very nice new code protocols for the alert levels.

Headmins (at least Ikarrus) agreed to have this added in, but Ikarrus wrote:
"Anything more involving than changing the current text strings we get needs to go through coderbus, not us."

### LINKED ISSUE #8068: Rev/cult and golems.
Right now when golem is created, it recieves this message:

You are an adamantine golem. You move slowly, but are highly resistant to heat and cold as well as blunt trauma. You are unable to wear clothes, but can still use most tools. Serve Ronnette Paynter, and assist them in completing their goals at any cost.

Golems can also be reved, which will result in cases like this:

Serve Ronnette Paynter, and assist them in completing their goals at any cost.

You are now a revolutionary! Help your cause. Do not harm your fellow freedom fighters. You can identify your comrades by the red "R" icons, and your leaders by the blue "R" icons. Help them kill the heads to win the revolution!

This is   Ronnette Paynter!
She is wearing   Ronnette Paynter's ID Card (Chief Engineer).

Say, non rev master orders golem to protect a head of staff. Antag status tells golem to kill said head of staff. Golem is now confused and wants to suicide becouse it has no idea which one takes precedence, antag status or masters orders.

Making golems unable to be reved, or giving them a notification that they no longer have to assist and listen to their master once they are brainwashed might help with the confusion.

### PR DESCRIPTION
**Pull request by ghost**
- Closes #7389 (The logging issues seem to have been fixed)
- Fixes #8068 (Adamantine golems are now automatically converted to the team antagonist that their master is currently a part of. They receive an additional message stating that their master still holds primary dominion over them and that they usually always will. Outside of their initial creation, adamantine golems can _not_ become team antagonists.

More fixes to come

### CODE REVIEW

File: code/modules/reagents/chemistry/holder.dm [hunk 1]
```diff
@@ -472,10 +472,11 @@ var/const/INJECT = 5 //injection
 
 	return 0
 
-/datum/reagents/proc/clear_reagents()
+/datum/reagents/proc/clear_reagents(exclude_id)
 	for(var/reagent in reagent_list)
 		var/datum/reagent/R = reagent
-		del_reagent(R.id)
+		if(R.id != exclude_id)
+			del_reagent(R.id)
```
Comment (phil235) [inline code/modules/reagents/chemistry/holder.dm]: This is a bad tradeoff, we're not going to add multiple checks to clear_reagents() just to fix a single issue.

**Comment by KorPhaeron** [comment]
>They receive an additional message stating that their master still holds primary dominion over them and that they usually always will.

Not a good idea. If I get converted to a cultist aftrr my boss tells me to kill cultists, I should stop killing cultists (especially now that I can see then all)

**Comment by Xhuis** [comment]
@KorPhaeron Golems are not convertable to team antagonists outside of initial creation. If the person who created them isn't an antagonist, they can't become one. I probably should have specified that.

File: code/modules/reagents/chemistry/holder.dm (hunk 1 above)
Comment (Xhuis) [inline code/modules/reagents/chemistry/holder.dm]: Any ideas for a better method? I'm all ears.

Comment (phil235) [inline code/modules/reagents/chemistry/holder.dm]: Yes, actually I have an idea: don't fix slime jelly in this PR, and let me fix the clusterfuck that is our way of handling blood, which will fix the slime jelly issue.

**Comment by Shadowlight213** [comment]
@Xhuis #7839 was closed over a year ago?

**Comment by Xhuis** [comment]
Mistype of #7389

File: code/modules/reagents/chemistry/holder.dm (hunk 1 above)
Comment (Xhuis) [inline code/modules/reagents/chemistry/holder.dm]: Good enough for me.

**Comment by optimumtact** [comment]
F I N I S H T H I S

**Comment by Xhuis** [comment]
I T ' S   F I N I S H E D
E X C E P T   F O R   T H A T   O N E   T H I N G
L E T   M E   F I X   T H A T

**Comment by Xhuis** [comment]
O K A Y Y O U ' R E G O O D

**Pull request merged**